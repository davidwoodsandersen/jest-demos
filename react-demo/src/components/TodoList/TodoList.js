import React from 'react';
import './TodoList.scss';
import Todo from '../Todo/Todo';

function TodoList(props) {
  const todos = props.todos ? props.todos.map(todo => {
    return (
      <Todo text={todo} key={Math.random() * 1000} />
    );
  }) : '';

  return (
    <div className="TodoList">
      <h3>Todo List</h3>
      {todos}
    </div>
  );
}

export default TodoList;
