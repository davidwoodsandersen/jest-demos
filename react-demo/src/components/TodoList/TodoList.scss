.TodoList {
  margin-bottom: 5px;
}
