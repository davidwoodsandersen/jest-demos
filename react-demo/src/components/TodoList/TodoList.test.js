import React from 'react';
import ReactDOM from 'react-dom';
import TodoList from './TodoList';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TodoList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const instance = renderer.create(<TodoList todos={['a', 'b', 'c']} />).toJSON();
  expect(instance).toMatchSnapshot();
});
