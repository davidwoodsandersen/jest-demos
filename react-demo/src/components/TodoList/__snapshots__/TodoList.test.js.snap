// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="TodoList"
>
  <h3>
    Todo List
  </h3>
  <div
    className="Todo"
  >
    a
  </div>
  <div
    className="Todo"
  >
    b
  </div>
  <div
    className="Todo"
  >
    c
  </div>
</div>
`;
